### PR TITLE
Support GM segment & vlan segment resources

### PR DESCRIPTION
**Pull request by asarfaty**

### CODE REVIEW

File: nsxt/resource_nsxt_policy_vlan_segment_test.go [hunk 1]
```diff
@@ -201,8 +201,15 @@ func testAccNsxtPolicyVlanSegmentExists(resourceName string) resource.TestCheckF
 		if resourceID == "" {
 			return fmt.Errorf("Policy VLAN Segment resource ID not set in resources")
 		}
+		var err error
```
Comment (enhaocui) [inline nsxt/resource_nsxt_policy_vlan_segment_test.go:204]: Shall we use resourceNsxtPolicySegmentExists here?

File: nsxt/resource_nsxt_policy_vlan_segment_test.go [hunk 2]
```diff
@@ -213,15 +220,21 @@ func testAccNsxtPolicyVlanSegmentExists(resourceName string) resource.TestCheckF
...
-	nsxClient := infra.NewDefaultSegmentsClient(connector)
 	for _, rs := range state.RootModule().Resources {
 
 		if rs.Type != "nsxt_policy_vlan_segment" {
 			continue
 		}
 
 		resourceID := rs.Primary.Attributes["id"]
-		_, err := nsxClient.Get(resourceID)
+		var err error
```
Comment (enhaocui) [inline nsxt/resource_nsxt_policy_vlan_segment_test.go:230]: Same here

File: nsxt/segment_common.go [hunk 3]
```diff
@@ -779,14 +814,29 @@ func nsxtPolicySegmentDelete(d *schema.ResourceData, m interface{}) error {
 		Pending: pendingStates,
 		Target:  targetStates,
 		Refresh: func() (interface{}, string, error) {
-			ports, err := portsClient.List(id, nil, nil, nil, nil, nil, nil)
-			if err != nil {
-				return ports, "error", logAPIError("Error listing segment ports", err)
+			var ports interface{}
+			var numOfPorts int
+			if isPolicyGlobalManager(m) {
+				portsClient := gm_segments.NewDefaultPortsClient(connector)
```
Comment (annakhm) [inline nsxt/segment_common.go:820]: I wonder if the port gets propagated to global manager?

File: nsxt/resource_nsxt_policy_vlan_segment_test.go (hunk 1 above)
Comment (asarfaty) [reply nsxt/resource_nsxt_policy_vlan_segment_test.go:204]: Done

File: nsxt/resource_nsxt_policy_vlan_segment_test.go (hunk 2 above)
Comment (asarfaty) [reply nsxt/resource_nsxt_policy_vlan_segment_test.go:230]: Done

File: nsxt/segment_common.go (hunk 3 above)
Comment (asarfaty) [reply nsxt/segment_common.go:820]: I'm not sure what you mean. You can create ports on the GM, so you should be able to wait for their deletion.

**Comment by asarfaty** [comment]
I'll add the additional test later

File: nsxt/segment_common.go (hunk 3 above)
Comment (annakhm) [reply nsxt/segment_common.go:826]: This code was added to deal with automatic port creation for vsphere vms. I'm not sure global manager will see these port. But this is something we can test later.

**Pull request merged**